dist/tools/lazysponge/lazysponge.py

```diff
@@ -0,0 +1,99 @@
+#! /usr/bin/env python3
+
+#
+# Copyright (C) 2018 Gaëtan Harter <gaetan.harter@fu-berlin.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+"""
+lazysponge
+
+Adaptation of moreutils `sponge` with added functionnality that it does not
+modify the output file if the content would be unchanged.
+
+Description
+-----------
+
+Reads standard input and writes it to the specified file if its content was
+different.
+
+The file is not changed if the content is the same so modification timestamp is
+unchanged.
+
+Note
+----
+
+It only works with input provided by a `pipe` and not interractive input.
+The reason is that `ctrl+c` would not be handled properly in that case.
+
+Usage
+-----
+
+    usage: lazysponge.py [-h] outfile
+
+    Soak up all input from stdin and write it to <outfile> if it differs from
+    previous content. If the content is the same, file is not modified.
+
+    positional arguments:
+    outfile     Output file
+
+    optional arguments:
+    -h, --help  show this help message and exit
+"""
+
+import os
+import sys
+import argparse
+import hashlib
+
+DESCRIPTION = ('Soak up all input from stdin and write it to <outfile>'
+               ' if it differs from previous content.\n'
+               ' If the content is the same, file is not modified.')
+PARSER = argparse.ArgumentParser(description=DESCRIPTION)
+PARSER.add_argument('outfile', help='Output file')
+PARSER.add_argument('--verbose', help='Verbose output', default=False,
+                    action='store_true')
+
+
+def _print_hash_debug_info(outfilename, oldbytes, newbytes):
+    """Print debug information on hashs."""
+    oldhash = hashlib.md5(oldbytes).hexdigest() if oldbytes is not None else ''
```

Comment:
It could have the same size and be different. I also kept the previous behavior as before.
And when trying changes a hash is easy enough to recognize for 5 runs in a row.